### PR TITLE
DOP-4941: Allow fonts for hidden languages to work without feature flag

### PR DESCRIPTION
**Pull request by rayangler**
### Stories/Links:

DOP-4941

### Current Behavior:

* [ja-jp](https://www.mongodb.com/ja-jp/docs/atlas/getting-started/) - Main issue: `Euclid Circular A` is being used as the font instead of `Noto Sans JP`. This is what's causing the Japanese characters to look way too big in comparison to English characters.
* [zh-cn](https://www.mongodb.com/zh-cn/docs/atlas/getting-started/)
* [ko-kr](https://www.mongodb.com/ko-kr/docs/atlas/getting-started/)
* [pt-br](https://www.mongodb.com/pt-br/docs/atlas/getting-started/)
* [English](https://www.mongodb.com/docs/atlas/getting-started/)

### Staging Links:

* [ja-jp](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/ja-jp/sandbox/cloud-docs/raymund.rodriguez/DOP-4941-localized-font-size/getting-started-jp/index.html) - Built with no feature flag to show that intended font will still be used, but Japanese will not appear as an option in the consistent-nav components. Japanese text is just for show.
* [zh-cn](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/zh-cn/sandbox/cloud-docs/raymund.rodriguez/DOP-4941-localized-font-size/getting-started/index.html)
* [ko-kr](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/ko-kr/sandbox/cloud-docs/raymund.rodriguez/DOP-4941-localized-font-size/getting-started/index.html)
* [pt-br](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/pt-br/sandbox/cloud-docs/raymund.rodriguez/DOP-4941-localized-font-size/getting-started/index.html)
* [English](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/sandbox/cloud-docs/raymund.rodriguez/DOP-4941-localized-font-size/getting-started/index.html)

**To test:**
1. Go to the Japanese staging link.
2. Ensure that Japanese does **not** show up in consistent-nav language dropdown. (This won't be the case for pre-prd since pre-prd has the feature flag turned on)
3. Ensure that `Noto Sans JP` font is loaded correctly on initial load.
4. Ensure Chinese and Korean sites load in their respective fonts correctly on initial load as well.

### Notes:

* Turns out the reason why font sizes were wonky for Japanese was because `Noto Sans JP` was not being used due to the lack of feature flag. This PR allows the use of a new `getAvailableLanguages` function that can get all languages, either through a feature flag or boolean. I have reached out to our Japanese translator, Saori, who has confirmed that the current font on preprd looks good.
* Fonts for different locales will now be initially loaded in without needing to check for the window's pathname. This prevents the flash of different font that is currently present on localized docs sites ([example](https://www.mongodb.com/zh-cn/docs/atlas/)).
* A new `GATSBY_LOCALE` env variable is now supported. This is used to help modify the `html` tag's `lang` attribute for local development purposes. The `lang` attribute is what the new CSS will use to help ensure correct fonts are used on initial load. Modifying the `lang` locally allows us to mimic how Smartling currently rewrites the `lang` attribute on localized sites. To use, add `GATSBY_LOCALE` to your env files and set it to a locale code.
* No major tests were included since it's a pretty visual change, and Japanese will no longer be a hidden language soon.

### README updates

- - [ ] This PR introduces changes that should be reflected in the README, and I have made those updates.
- - [x] This PR does not introduce changes that should be reflected in the README

### CODE REVIEW

File: src/components/Sidenav/Sidenav.js [hunk 1]
```diff
@@ -248,10 +239,10 @@ const Sidenav = ({ chapters, guides, page, pageTitle, repoBranches, slug, eol })
     <>
       <Global
         styles={css`
-          ${disableScroll(!hideMobile)} ${translatedFontFamilyStyles}
+          ${disableScroll(!hideMobile)}
         `}
       />
-      <SidenavContainer {...topValues} template={template}>
+      <SidenavContainer {...topValues} template={template} id={'side-nav-container'}>
```
Comment (mmeigs) [inline src/components/Sidenav/Sidenav.js:245]: I think this id is used in `Sidenav.test.js` (barely)

**Review by mmeigs — COMMENTED** [review]
One comment below. Also, idk why but none of the staging links above seem to be in their language except for Japanese? (this nav is, but the content is not)

File: src/components/Sidenav/Sidenav.js (hunk 1 above)
Comment (rayangler) [reply src/components/Sidenav/Sidenav.js]: Good catch. I went ahead and changed it to `docs-side-nav-container` to help differentiate. Struggled with coming up with a better name, so I'll happily accept other suggestions 😄

**Review by rayangler — COMMENTED** [review]
> none of the staging links above seem to be in their language except for Japanese? (this nav is, but the content is not)

Yeah, this is intentional. The Japanese staging link points to a completely new page I created and copied Japanese content into the txt file. I can do that for the other languages too if that helps with visualizing the changes. Unfortunately, I'm not sure how else to mock translated content from Smartling with the current setup

**Review by mayaraman19 — APPROVED** [review]
this is cool! I left a couple comments but looks good to me, and all the staging sites have the correct fonts 👍

File: src/constants.js [hunk 2]
```diff
@@ -1,3 +1,7 @@
+// Append "docs" to the beginning due to differentiate from LG's internal side-nav-container testid
```
[the remark this answers is not in the document]
Comment (rayangler) [reply src/constants.js]: Updated the comment to include that it's specficially for the side nav one. Let me know if that helps, or if there's a preferred alternative to handle this!

File: src/utils/locale.js [hunk 3]
```diff
@@ -19,17 +20,30 @@ export const STORAGE_KEY_PREF_LOCALE = 'preferredLocale';
 // in redirect-based-on-lang.js
 export const AVAILABLE_LANGUAGES = [
```
[the remark this answers is not in the document]
Comment (rayangler) [reply src/utils/locale.js]: Thanks for catching this! Done

**Review by mmeigs — APPROVED** [review]

**Pull request merged**